### PR TITLE
fix(gemma4): Gemma4 packing, attention mask, and fixesMoE routing

### PR DESCRIPTION
**Pull request by shruthan**
# What does this PR do ?

Fixes some bugs in VLM/Gemma4 SFT with packing. 

# Changelog

- `neat_packing_vlm.py`: propagate `mm_token_type_ids` and `image_position_ids` through packing pipeline
- `collate_fns.py`: include `mm_token_type_ids` in `neat_packed_vlm_collater` output
- `gemma4_moe/model.py`: convert 4D bool attention mask to additive format before eager attention
- `gemma4_moe/model.py`: select top-k experts from `router_probs` (softmax) instead of raw `expert_scores`, consistent with HF Gemma4 implementation

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Automodel/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by HuiyingLi** [comment]
Thank you @shruthan ! Would you mind also adding a packed sequence config yaml when you get chance?

**Comment by athitten** [comment]
The code LGTM, thank you @shruthan for your contribution! Also did a quick verification of the loss curve with packing and it looks good:

Screenshot 2026-05-05 at 5 17 06 PM

**Review by athitten — DISMISSED** [review]
Approving, thank you!!

**Comment by shruthan** [comment]
> Approving, thank you!!

Thanks @athitten . I force pushed with sign-off which was blocking earlier, might need to approve again.

**Review by HuiyingLi — APPROVED** [review]
